core/src/main/java/feast/core/controller/exception/handler/RestResponseEntityExceptionHandler.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.controller.exception.handler;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class RestResponseEntityExceptionHandler extends ResponseEntityExceptionHandler {
+
+  @ExceptionHandler(value = {InvalidProtocolBufferException.class})
+  protected ResponseEntity<Object> handleInvalidProtocolBufferException(
+      RuntimeException ex, WebRequest request) {
+    String bodyOfResponse = "An invalid response was received from Feast Core.";
```

Comment:
Seems like "Unexpected error occurred in Feast Core" might make more sense.
Also this should be returned as JSON instead of a plain string as the client has been talking to Feast Core in JSON all along using JSON.